File: library/src/main/java/com/a65apps/library/database/dao/LocationDao.java
package com.a65apps.library.database.dao;

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.a65apps.library.models.LocationModel;

import java.util.List;
@Dao
public interface LocationDao {
    @Query("SELECT * FROM LocationModel")
    @NonNull
    List<LocationModel> getAll();

    @Query("SELECT * FROM LocationModel WHERE person_id = :id")
    @NonNull
    LocationModel getByPersonId(@NonNull String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertLocation(@NonNull LocationModel location);

    @Query("SELECT EXISTS(SELECT * FROM LocationModel WHERE person_id = :id)")
    int isExists(@NonNull String id);

    @Delete
    void delete(LocationModel locationModel);
}

File: library/src/main/java/com/a65apps/library/database/dao/LocationDao.kt
package com.a65apps.library.database.dao

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.a65apps.library.models.LocationModel

interface LocationDao {
    @Query("SELECT * FROM LocationModel")
    fun getAll(): List<LocationModel>

    @Query("SELECT * FROM LocationModel WHERE person_id = :id")
    fun getByPersonId(id: String): LocationModel

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertLocation(location: LocationModel)

    @Query("SELECT EXISTS(SELECT * FROM LocationModel WHERE person_id = :id)")
    fun isExist(id: String): Int

    @Delete
    fun delete(locationModel: LocationModel?)
}